pages/stands/index.tsx

```diff
@@ -0,0 +1,299 @@
+    stands_QueryResponse['stands']
+  >([]);
+
+  const {
+    mainCollaborator,
+    collaborators,
+    companiesFirstDay,
+    companiesLastDay,
+    startDate,
+    endDate,
+  } = props.currentMetaData;
+
+  useEffect(() => {
+    const interval = setInterval(() => setTime(dayjs()), intervalLength);
+    return (): void => clearInterval(interval);
+  }, [props.currentMetaData]);
+
+  useEffect(() => {
+    const updatedFeaturedStands = getFeaturedEventStands(
+      time,
+      props.stands,
+      startDate,
+      endDate,
+      companiesFirstDay,
+      companiesLastDay
+    );
+    if (updatedFeaturedStands) {
+      setFeaturedEventStands(updatedFeaturedStands);
+    }
+  }, [
+    time,
+    props.stands,
+    startDate,
+    endDate,
+    companiesFirstDay,
+    companiesLastDay,
+  ]);
+
+  return timeIsAfter({
+    time: time,
+    start: toDayjs(props.currentMetaData.startDate, '09:30:00'),
+  }) ? (
+    <StandsDefault currentMetaData={props.currentMetaData} />
+  ) : (
+    <>
+      {props.stands_page && (
+        <PageView hideContent hideDate hideTitle page={props.stands_page} />
+      )}
+
+      {/* TODO: Complete technical implementation of the LivePlayer */}
+      <LivePlayer livestreamUrl={''} qaUrl={''} />
+      <PaddedDivider />
+
+      {props.stands &&
+        featuredEventStands &&
+        featuredEventStands.length > 0 && (
+          <FeaturedEvents time={time} stands={featuredEventStands} />
+        )}
+
+      {mainCollaborator && (
+        <HSPGrid>
+          {props.stands
+            ?.filter(
+              (stand) =>
+                stand &&
+                currentDayCompaniesIds(
+                  startDate,
+                  endDate,
+                  companiesFirstDay,
+                  companiesLastDay
+                ).includes(stand.company.id) &&
+                stand.company.id === mainCollaborator.id
+            )
+            .map((stand) => (
+              <StandCard
+                stand={stand}
+                key={mainCollaborator.id}
+                time={time}
+                type={'hsp'}
+              />
+            ))}
+        </HSPGrid>
+      )}
+
+      <SPGrid>
+        {props.stands
+          ?.filter(
+            (stand) =>
+              stand &&
+              currentDayCompaniesIds(
+                startDate,
+                endDate,
+                companiesFirstDay,
+                companiesLastDay
+              ).includes(stand.company.id) &&
+              companyIds(collaborators).includes(stand.company.id)
+          )
+          .map((stand) => (
+            <StandCard key={stand.id} stand={stand} time={time} type={'sp'} />
+          ))}
+      </SPGrid>
+
+      <StandGrid>
+        {props.stands
+          ?.filter(
+            (stand) =>
+              stand &&
+              currentDayCompaniesIds(
+                startDate,
+                endDate,
+                companiesFirstDay,
+                companiesLastDay
```

Comment:
This function is called a lot of times here. We should just call it once and use the resulting array in these places instead.

Reply:
Agreed, I'll create a state and populate it on mount

Reply:
You shouldn't need any state either :smile: